state-chain/pallets/cf-elections/src/electoral_systems/nonce_wintessing.rs

```diff
@@ -107,18 +102,35 @@ impl<
 		}
 	}
 
+	fn is_vote_valid<ElectionAccess: ElectionReadAccess<ElectoralSystem = Self>>(
+		_election_identifier: ElectionIdentifierOf<Self>,
+		election_access: &ElectionAccess,
+		partial_vote: &<Self::Vote as VoteStorage>::PartialVote,
+	) -> Result<bool, CorruptStorageError> {
+		let (_, previous_value, previous_slot) = election_access.properties()?;
+		Ok(partial_vote.value != previous_value && partial_vote.slot > previous_slot)
+	}
+	fn generate_vote_properties(
+		_election_identifier: ElectionIdentifierOf<Self>,
+		_previous_vote: Option<(VotePropertiesOf<Self>, AuthorityVoteOf<Self>)>,
+		_vote: &<Self::Vote as VoteStorage>::PartialVote,
+	) -> Result<VotePropertiesOf<Self>, CorruptStorageError> {
+		Ok(())
+	}
+
 	fn on_finalize<ElectoralAccess: ElectoralWriteAccess<ElectoralSystem = Self>>(
 		electoral_access: &mut ElectoralAccess,
 		election_identifiers: Vec<ElectionIdentifierOf<Self>>,
 		_context: &Self::OnFinalizeContext,
 	) -> Result<Self::OnFinalizeReturn, CorruptStorageError> {
 		for election_identifier in election_identifiers {
 			let mut election_access = electoral_access.election_mut(election_identifier)?;
-			if let Some(value) = election_access.check_consensus()?.has_consensus() {
-				let (identifier, previous_value) = election_access.properties()?;
-				if previous_value != value {
+			if let Some((value, slot)) = election_access.check_consensus()?.has_consensus() {
+				let (identifier, previous_value, previous_slot) = election_access.properties()?;
+				if previous_value != value && slot > previous_slot {
```

Comment:
Here we have the on-chain logic, which should prevent it on its own, without having to rely on correct behaviour from the engines. So I don't mind having this here as a way of ensuring the chain logic is valid independent of engine behaviour - though I do think there should be a comment here that we don't expect this to be false because in general the engines should behave.

Reply:
Yes I'm saying we remove it, just that add something so that we don't silently fail if the assumption is wrong.